### ISSUE TITLE
Reviewer doesn't show up unless role is removed and then added back

### ISSUE THREAD

**Issue by alexxxmendonca** [pkp/pkp-lib#6317]
**Describe the bug**
Sometimes a reviewer will not show up when searching for reviewers within a manuscript even though said user has the "Reviewer" role enabled.

To fix it, the manager needs to edit the user's profile, uncheck the "Reviewer" role, save, edit again the user's profile, check back the "Reviewer" role, save, go back to the submission and search again for the same reviewer.

**To Reproduce**
Step 1: can't find the reviewer even though he has the "Reviewer" role
![Untitled_ Nov 4, 2020 3_49 PM](https://user-images.githubusercontent.com/4059679/98155665-63f75700-1eb5-11eb-809d-261dc9cfbece.gif)

Step 2: search for the user and manually remove and add back again the "Reviewer" role (please be patient with this GIF, as it's 39 seconds long)
![Untitled_ Nov 4, 2020 3_51 PM (2)](https://user-images.githubusercontent.com/4059679/98156649-dae11f80-1eb6-11eb-920a-503a69401f7c.gif)

Step 3: go back to the submission and search for the reviewer again. He will now appear.
![Untitled_ Nov 4, 2020 3_55 PM](https://user-images.githubusercontent.com/4059679/98156205-3f4faf00-1eb6-11eb-9c78-c6581f3fe3c2.gif)

**What application are you using?**
OJS 3.2.1.1

**Additional information**
This happens with only certain cases. In the past, in other examples, we were able to identify that the user had previously been merged with another account. I was not able to tell if that was the case for this particular example.

This gets particularly challenging when the journal is unabled to edit the user's profile because said user is shared with other jorunals. This results is us (Site Admins) having to do this for them, making the process longer and frustrating for the journal.

**Comment by NateWr** [comment]
The challenge will be to find a way for us reproduce the problem. Until we can reproduce it we can't fix it. Can you identify steps that lead to this problem, such as merging a user? If you can find a set of steps that we can take, like creating a user then merging them, with specific values in the user fields before or after merge, then that will help us reproduce it.

**Comment by NateWr** [comment]
Oh, I think I see what's happening. In your first video, when no results are found, you should see text indicating that:

![reviewer-error](https://user-images.githubusercontent.com/2306629/98229304-a8d0cb80-1f51-11eb-8c59-4f7b1a89448b.png)

Instead you see nothing appear there. That means that there is an error happening here. Usually, when an item in a list of items (like one reviewer in a list) does not appear, that's because of a JavaScript error. You can look at your browser's console log to see if you see an error there.

But still the best thing to do is still to identify a series of steps we can take to reproduce it on our end. That way we can find and resolve the JavaScript error locally.

**Comment by alexxxmendonca** [comment]
Nice catch!

Given that we only know something like this happens when a journal contact us, there is no way right now to simulate a scenario in which the error can be reproduced.

For this particular case, it's been fixed the manual way so that the journal could move on with their tasks.

However we will try what you suggested the next time this comes to our attention.

**Comment by NateWr** [comment]
@alexxxmendonca you could try to reproduce this by creating a fake reviewer and merging that user into another user. Or however it was that the merge happens that leads to the problem. Basically if you can find a reliable set of steps that _leads to_ a reviewer search not working. In this case, by the time someone reports a bug to you, the problem has already been "produced" and it may not be possible to determine how it was produced.

**Comment by alexxxmendonca** [comment]
The merging thing is just a hunch/guess. I am not sure if it's related at all.

We could not find records of that particular user being previously merged, for example. We know that because most of the times journals request our help by e-mail to merge users, given that they cannot merge themselves if one of the accounts is enrolled to another journal. 

No records of previous merge for that particular user have been found.

**Comment by alexxxmendonca** [comment]
@NateWr Just for the sake of it, I did try to simulate something on a test journal and test users.

1) Reviewer A and B were both findable in the reviewer search results.
2) Reviewer A and B were merged, maintaining Reviewer A as the remaining account.
3) Reviewer A could be found in the reviewer search results.

In summary, I could not conclude my theory that there is any relation with previous merging.

**Comment by NateWr** [comment]
Thanks for checking! I guess the problem is most likely a missing piece of data about the reviewer. If you want to continue testing, you could try removing data about the reviewer field by field, testing as you go, to see if you can reproduce it and which field was missing. Otherwise we can wait for it to surface again.

**Comment by alexxxmendonca** [comment]
Hi @NateWr 

So another case popped up with the exact same symptoms.

Here's what the browser's console log says:

![image](https://user-images.githubusercontent.com/4059679/98806180-88908900-23f7-11eb-89be-4b611b92372d.png)

```
DevTools failed to load SourceMap: Could not load content for https://submission.scielo.br/lib/pkp/lib/vendor/tinymce/tinymce/skins/lightgray/skin.min.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://submission.scielo.br/plugins/themes/default/js/lib/popper/popper.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://submission.scielo.br/plugins/themes/default/js/lib/bootstrap/util.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://submission.scielo.br/plugins/themes/default/js/lib/bootstrap/dropdown.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
jquery.min.js?v=3.2.1.1:2 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
send @ jquery.min.js?v=3.2.1.1:2
ajax @ jquery.min.js?v=3.2.1.1:2
a.pkp.controllers.NotificationHandler.fetchNotificationHandler_ @ pkp.min.js?v=3.2.1.1:225
a.pkp.classes.Handler.handleEvent @ pkp.min.js?v=3.2.1.1:129
dispatch @ jquery.min.js?v=3.2.1.1:2
y.handle @ jquery.min.js?v=3.2.1.1:2
trigger @ jquery.min.js?v=3.2.1.1:2
triggerHandler @ jquery.min.js?v=3.2.1.1:2
a.pkp.controllers.NotificationHandler @ pkp.min.js?v=3.2.1.1:224
a.pkp.classes.ObjectProxy.parent @ pkp.min.js?v=3.2.1.1:124
b @ pkp.min.js?v=3.2.1.1:122
a.pkp.classes.Helper.objectFactory @ pkp.min.js?v=3.2.1.1:120
(anonymous) @ pkp.min.js?v=3.2.1.1:429
each @ jquery.min.js?v=3.2.1.1:2
each @ jquery.min.js?v=3.2.1.1:2
a.fn.pkpHandler @ pkp.min.js?v=3.2.1.1:429
(anonymous) @ 3:208
l @ jquery.min.js?v=3.2.1.1:2
c @ jquery.min.js?v=3.2.1.1:2
setTimeout (async)
(anonymous) @ jquery.min.js?v=3.2.1.1:2
u @ jquery.min.js?v=3.2.1.1:2
fireWith @ jquery.min.js?v=3.2.1.1:2
fire @ jquery.min.js?v=3.2.1.1:2
u @ jquery.min.js?v=3.2.1.1:2
fireWith @ jquery.min.js?v=3.2.1.1:2
ready @ jquery.min.js?v=3.2.1.1:2
_ @ jquery.min.js?v=3.2.1.1:2
DevTools failed to load SourceMap: Could not load content for https://submission.scielo.br/lib/pkp/lib/vendor/tinymce/tinymce/skins/lightgray/skin.min.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
build.js?v=3.2.1.1:14 TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at o.ta (build.js?v=3.2.1.1:8)
    at o.e._render (build.js?v=3.2.1.1:14)
    at o.i (build.js?v=3.2.1.1:14)
    at ri.get (build.js?v=3.2.1.1:14)
    at new ri (build.js?v=3.2.1.1:14)
    at In (build.js?v=3.2.1.1:14)
    at o.wi.$mount (build.js?v=3.2.1.1:14)
    at o.wi.$mount (build.js?v=3.2.1.1:14)
    at init (build.js?v=3.2.1.1:14)
```

**Comment by NateWr** [comment]
Thanks! It sounds like there is some missing data in the return object for that reviewer. Before fixing this, can you do the following to help identify the missing property.

1. Go to the Console in Firefox.
2. Open the Network tab (it should be along the same line of options as the Console)
3. With the Network Tab open, select XHR.
4. With the Network Tab open, go to the add reviewer area.
5. Search for the reviewer.
6. In the network tab, you should see a new request appear when the search is made. Right-click this request and then select Copy > Copy Response.
7. Copy that response to a .txt file and then share that with me in slack. (Don't share it here because it will contain private details about that reviewer)

**Comment by NateWr** [comment]
Thanks @alexxxmendonca, that helped! It looks like this is caused by code that expects the `affiliation` to be entered and when there is none would cause an error. This was identified and fixed in https://github.com/pkp/pkp-lib/issues/6151.

It looks like that was just a couple months before 3.2.1-1 was released. The fix should arrive with 3.2.1-2, which we expect to come out in the next week or two.

In the meantime, you can fix this by entering an affiliation for the reviewer (or maybe just saving the edit user form without entering an affiliation).

**Comment by NateWr** [comment]
Sorry, I meant a couple months _after_ 3.2.1-1 was released. So the fix was not included in 3.2.1-1 but will be in 3.2.1-2.

**Comment by alexxxmendonca** [comment]
Thanks a lot for all your help @NateWr !
